.github/workflows/static_analysis.yml

```diff
@@ -17,13 +17,14 @@ concurrency:
 
 env:
   COMPILER: 'gcc'
+  MAKEFLAGS: '-j4'
```

Comment:
While looking at this configuration, I realized that the `check-r-docs` job is installing a bunch of R packages on Linux. CRAN doesn't provide Linux binaries, so that means they're all being compiled from source.

Parallelizing compilation should speed that job up, as we do with the other R jobs:

https://github.com/microsoft/LightGBM/blob/e7edb6cb1894f0d3847c3eaf3df9fc5a6b2414f9/.github/workflows/r_package.yml#L26